store/src/main/java/com/zimbra/cs/service/admin/IssueCert.java

```diff
@@ -0,0 +1,115 @@
+ * {@link com.zimbra.cs.rmgmt.RemoteManager}, {@link RemoteCertbot}.
+ *
+ * @author Yuliya Aheeva
+ * @since 23.3.0
+ */
+public class IssueCert extends AdminDocumentHandler {
+
+  /**
+   * Handles the request. Searches a domain by id, checks admin rights (accessible to global and
+   * delegated admin of requested domain), searches a server with proxy node, executes remote
+   * certbot command on it, creates response element.
+   *
+   * @param request {@link Element} representation of {@link
+   *     com.zimbra.soap.admin.message.IssueCertRequest}
+   * @param context request context
+   * @return {@link Element} representation of {@link
+   *     com.zimbra.soap.admin.message.IssueCertResponse}
+   * @throws ServiceException in case if domain could not be found, domain doesn't have
+   * PublicServiceHostname and at least one VirtualHostName,
+   * server with proxy node could not be found or domain admin dosn't have rights to deal with
+   * this domain.
+   *
+   * It won't throw an exception in case if remote execution command fails, instead will create
+   * a response and add a failure message to it.
+   */
+  @Override
+  public Element handle(final Element request, final Map<String, Object> context)
+      throws ServiceException {
+
+    ZimbraSoapContext zsc = getZimbraSoapContext(context);
+
+    Provisioning prov = Provisioning.getInstance();
+    String domainId = request.getAttribute(AdminConstants.A_DOMAIN);
+    Domain domain = Optional.ofNullable(prov.get(DomainBy.id, domainId))
+        .orElseThrow(() -> ServiceException.INVALID_REQUEST(
+        "Domain with id " + domainId + " could not be found.", null));
+
+    AdminAccessControl admin = checkDomainRight(zsc, domain, AdminRights.R_getDomain);
+    String adminMail = admin.mAuthedAcct.getMail();
+
+    String chain = request.getAttribute(AdminConstants.A_CHAIN_TYPE, AdminConstants.DEFAULT_CHAIN);
+    boolean expand = request.getAttributeBool(MailConstants.A_EXPAND, false);
+
+    String publicServiceHostname = domain.getPublicServiceHostname();
+    String[] virtualHostNames = domain.getVirtualHostname();
+    if (publicServiceHostname == null) {
+      throw ServiceException.FAILURE(
+          "Domain with id " + domainId + " must have PublicServiceHostname.",
+          null);
+    } else if (virtualHostNames.length == 0) {
+      throw ServiceException.FAILURE(
+          "Domain with id " + domainId + " must have at least one VirtualHostName.",
+          null);
+    }
+
+    // First release will work only on ONE proxy even if there are multiple proxy in the infrastructure.
+    Server proxyServer =
+        prov.getAllServers()
+            .stream()
+            .filter(Server::hasProxyService)
+            .findFirst()
+            .orElseThrow(() -> ServiceException.FAILURE(
+                "Issuing LetsEncrypt certificate command requires carbonio-proxy node. "
+                + "Be sure carbonio-proxy is installed, up and running."));
```

Comment:
Consider changing to:
```suggestion
                "Issuing LetsEncrypt certificate command requires a carbonio-proxy node. "
                + "Make sure a carbonio-proxy node is installed, up and running in your infrastructure."));
```

Reply:
Also, we dont need to specify "node" keyword. The install could be single server and maybe for some weird reasons server could have missing "proxy" in enabled services.

Reply:
Ok then just 
```suggestion
                "Issuing LetsEncrypt certificate command requires carbonio-proxy. "
                + "Be sure carbonio-proxy is installed, up and running."));
```
should be fine :slightly_smiling_face: 

or

```suggestion
                "Issuing LetsEncrypt certificate command requires carbonio-proxy service to be available. "
                + "Be sure carbonio-proxy is installed, up and running."));
```